src/BotPlutusInterface/Balance.hs

```diff
@@ -527,21 +536,34 @@ modifyFirst ::
 modifyFirst _ m [] = m Nothing `consJust` []
 modifyFirst p m (x : xs) = if p x then m (Just x) `consJust` xs else x : modifyFirst p m xs
 
-minus :: Value -> Value -> Value
-minus x y =
+minus :: CApi.Value -> CApi.Value -> CApi.Value
+minus a b = a <> CApi.negateValue b
+
+ledgerMinus :: Value -> Value -> Value
+ledgerMinus x y =
   let negativeValues = map (\(c, t, a) -> (c, t, - a)) $ Value.flattenValue y
    in x <> mconcat (map unflattenValue negativeValues)
 
 unflattenValue :: (CurrencySymbol, TokenName, Integer) -> Value
 unflattenValue (curSymbol, tokenName, amount) =
   Value.assetClassValue (Value.assetClass curSymbol tokenName) amount
```

Comment:
If `flattenValue :: Value -> [(CurrencySymbol, TokenName, Integer)]`, this should be renamed. I would expect `unflattenValue :: [(CurrencySymbol, TokenName, Integer)] -> Value`.  

BTW it is `uncurry3 Value.singleton`

Reply:
So it is, we can blame Gergely from 13 months ago lol